Firmware/targets/arm/Src/HAL.cpp

```diff
@@ -95,12 +97,12 @@ void setCallback(void (*callback_)(char* data)) {
 
 namespace Encoder {
 int getCountAndReset() {
-    return 0;
+    return encoderGetAndReset();
 }
 void setButtonCallback(void (*callback)()) {
-    UNREFERENCED_PARAMETER(callback);
+    encoderSetCallback(callback);
```

Comment:
encoderSetButtonCallback for now. in future maybe separate button driver to other file